### ISSUE TITLE
Reading of yaml files and usage of astropy constructors

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/simtools#812]
We usually use `astropy.io.misc.yaml` to read yaml files. 

This has the disadvantage that it is much slower to import (numbers on my local computer):

- 396388    astropy.io.misc.yaml
- 32077   yaml

These are us, so we are talking about a factor of 12! This is roughly 1/3 of the import time of general.py.

Question: do we actual use the yaml constructors for astropy anywhere? My feeling is that they are nice, but not needed.

That you don't have to search - here an example, everything starting with `!`:

```
enith_angle:
  len: 1
  unit: !astropy.units.Unit {unit: deg}
  default: !astropy.units.Quantity
    value: 20
    unit: !astropy.units.Unit {unit: deg}
  names: ['zenith', 'theta']
azimuth_angle:
  len: 1
  unit: !astropy.units.Unit {unit: deg}
  default: !astropy.units.Quantity
    value: 0
    unit: !astropy.units.Unit {unit: deg}
  names: ['azimuth', 'phi']
nsb_spectrum:
  len: 1
  unit: null
  default: null
```

I suggest to remove those constructors and use plain yaml. Please protest if you don't agree.

**Comment by orelgueta** [comment]
I assume you suggest to instead write something like
```
value: 20
unit: deg
```
read it with "plain" yaml and then construct a unit as `zenith = u.Quantity(value, unit)`. Is that correct? Did you check it is faster than importing `astropy yaml`?

**Comment by GernotMaier** [comment]
I didn't test it: if `zenith = u.Quantity(value, unit)` takes 0.4 us, then we have a lot of reasons to drop astropy completely. Do you want me to do that? I really think we should go for the simpler approach and depend on astropy only where absolutely necessary.

**Comment by orelgueta** [comment]
OK, so to test it, I ran two scripts:
- importing astropy yaml, reading the file you pasted above and accessing the default value.
- importing plain yaml and astropy units in order to construct a quantity.

The code is given below.

The results are as follows:
Average execution time for plain yaml in 100 runs: 162.09767791999997 ms
Average execution time for astropy yaml in 100 runs: 276.02657125 ms

Either code is extremely slow anyway cause we import units. The difference between them is less than a factor two, so changing from one to the other would help, but not that much.

```
import astropy.units as u
import yaml

data_yaml = yaml.safe_load(open("test_yaml.yml"))
zenith_angle_yaml = u.Quantity(
    data_yaml["zenith_angle"]["default"]["value"], data_yaml["zenith_angle"]["default"]["unit"]
)
```

```
import astropy.io.misc.yaml as astro_yaml

data_astro = astro_yaml.load(open("test_yaml_from_astropy.yml"))
zenith_angle_astropy = data_astro["zenith_angle"]["default"]
```

```
import timeit
import subprocess


def run_plain_yaml():
    # Run using a shell command the python program plain_yaml.py
    subprocess.run(["python", "plain_yaml.py"])


def run_astropy_yaml():
    # Run using a shell command the python program astropy_yaml.py
    subprocess.run(["python", "astropy_yaml.py"])


n_runs = 100
# Time the execution of the run_script function
execution_time = timeit.timeit(run_plain_yaml, number=n_runs)

print(
    f"Average execution time for plain yaml in {n_runs} runs: {execution_time / n_runs / 1e-3} ms"
)

execution_time = timeit.timeit(run_astropy_yaml, number=n_runs)

print(
    f"Average execution time for astropy yaml in {n_runs} runs: {execution_time / n_runs /1e-3} ms"
)
```

**Comment by GernotMaier** [comment]
OK, thanks for this. Didn't expect you to do that.

It is definitely an improvement! But I agree that it is still not fast.

**Comment by GernotMaier** [comment]
This has been addressed in #813
